x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_host_unisolation.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useState } from 'react';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { HOST_ISOLATION_FAILURE } from './translations';
+import { createHostUnIsolation } from './api';
+
+interface HostUnisolationStatus {
+  loading: boolean;
+  unIsolateHost: () => Promise<boolean>;
+}
+
+interface UseHostIsolationProps {
+  agentId: string;
+  comment: string;
+  caseIds?: string[];
+}
+
+export const useHostUnisolation = ({
+  agentId,
+  comment,
+  caseIds,
+}: UseHostIsolationProps): HostUnisolationStatus => {
+  const [loading, setLoading] = useState(false);
+  const { addError } = useAppToasts();
+
+  const unIsolateHost = useCallback(async () => {
```

Comment:
Should this function also do the stuff around checking if the component is mounted?

Reply:
i guess so? i wasn't sure if we needed it as much because we are not saving the response in state and just sending it back directly. what do you think?